quickwit-indexing/src/actors/packager.rs

```diff
@@ -174,35 +174,6 @@ impl Handler<IndexedSplitBatch> for Packager {
 }
 
 /// returns true iff merge is required to reach a state where
-/// we have zero, or a single segment with no deletes segment.
-fn is_merge_required(segment_metas: &[SegmentMeta]) -> bool {
-    match &segment_metas {
-        // there are no segment to merge
-        [] => false,
-        // if there is only segment but it has deletes, it
-        // still makes sense to merge it alone in order to remove deleted documents.
-        [segment_meta] => segment_meta.has_deletes(),
```

Comment:
I'm surprised to see this condition on delete in the packager as this was coded before the deletes.

Reply:
that was defensive...